load/load_windows.go

```diff
@@ -37,20 +37,30 @@ func loadAvgGoroutine() {
 
 	counter, err := common.ProcessorQueueLengthCounter()
 	if err != nil || counter == nil {
-		log.Println("gopsutil: unexpected processor queue length counter error, please file an issue on github: err")
+		log.Printf("unexpected processor queue length counter error, %v\n", err)
 		return
 	}
 
 	tick := time.NewTicker(samplingFrequency).C
-	for {
+
+	f := func() {
 		currentLoad, err = counter.GetValue()
-		loadAvgMutex.Lock()
 		loadErr = err
+		loadAvgMutex.Lock()
 		loadAvg1M = loadAvg1M*loadAvgFactor1M + currentLoad*(1-loadAvgFactor1M)
 		loadAvg5M = loadAvg5M*loadAvgFactor5M + currentLoad*(1-loadAvgFactor5M)
 		loadAvg15M = loadAvg15M*loadAvgFactor15M + currentLoad*(1-loadAvgFactor15M)
 		loadAvgMutex.Unlock()
-		<-tick
+	}
+
+	f() // run first time
```

Comment:
go f() maybe better?

Reply:
Thank you for the comment. However, I don't think so. This `f()` is defined because I do not want to write the same function twice, the first time and called by tick (I want `do-while`). This function itself is in a goroutine, and the time to execute `counter.GetValue()` is very little, so I do not think it is necessary to increase a goroutine.